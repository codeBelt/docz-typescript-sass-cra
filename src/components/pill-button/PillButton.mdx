---
name: PillButton
---

import { Playground, Props } from 'docz'
import PillButton from './PillButton'

# PillButton

<Props of={PillButton} />

## Basic usage

<Playground>
  <PillButton>Click me</PillButton>
  <PillButton isActive="false">Click me</PillButton>
</Playground>
