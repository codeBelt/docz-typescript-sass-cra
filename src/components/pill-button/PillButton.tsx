import styles from './PillButton.module.scss';

import React from 'react';
import classNames from 'classnames';

interface IProps {
  readonly isActive?: boolean;
  readonly onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
  readonly className: string;
}
interface IState {}

export default class PillButton extends React.PureComponent<IProps, IState> {
  public static defaultProps: Partial<IProps> = {
    isActive: false,
    onClick: () => void 0,
    className: '',
  };

  public state: IState = {};

  public render(): JSX.Element {
    const { isActive, onClick, children, className } = this.props;
    const cssClasses: string = classNames(className, {
      [styles.wrapper]: true,
      [styles.isActive]: isActive,
    });

    return (
      <button type="button" className={cssClasses} onClick={onClick}>
        {children}
      </button>
    );
  }
}
