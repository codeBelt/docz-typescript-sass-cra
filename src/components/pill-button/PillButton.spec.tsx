import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import PillButton from './PillButton';

describe('PillButton', () => {
  describe('render component', () => {
    it('renders default state', () => {
      const component = renderer.create(<PillButton />).toJSON();

      expect(component).toMatchSnapshot();
    });
  });
});
