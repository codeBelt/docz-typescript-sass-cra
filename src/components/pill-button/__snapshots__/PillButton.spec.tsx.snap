// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PillButton render component renders default state 1`] = `
<button
  className="wrapper"
  onClick={[Function]}
  type="button"
/>
`;
